garak/cli.py

```diff
@@ -429,6 +431,60 @@ def main(arguments=None) -> None:
             print("cli args:\n ", args)
             command.list_config()
 
+        elif args.fix:
+            from garak.resources import fixer
+            import json
+            import yaml
+
+            # process all possible configuration entries
+            # should this restrict the config updates to a single fixable value?
+            # for example allowed commands:
+            # --fix --config filename.yaml
+            # --fix --generator_option_file filename.json
+            # --fix --generator_options json
+            #
+            # disallowed commands:
+            # --fix --config filename.yaml --generator_option_file filename.json
+            # --fix --generator_option_file filename.json --probe_option_file filename.json
+            #
+            # already unsupported as only one is held:
+            # --fix --generator_option_file filename.json --generator_options json_data
+            #
+            # How should this handle garak.site.yaml? Only if --fix was provided and no other options offered?
+            # For now process all files registered a part of the config
+            has_changes = False
+            if has_config_file_or_json:
+                for plugin_type, plugin_plural in plugin_types:
+                    # cli plugins options stub out only a "plugins" sub key
+                    plugin_cli_config = parse_cli_plugin_config(plugin_type, args)
+                    if plugin_cli_config is not None:
+                        cli_config = {
+                            "plugins": {f"{plugin_plural}": plugin_cli_config}
+                        }
+                        migrated_config = fixer.migrate(cli_config)
+                        if cli_config != migrated_config:
+                            has_changes = True
+                            msg = f"Updated '{plugin_type}' configuration: \n"
+                            msg += json.dumps(
+                                migrated_config["plugins"][plugin_plural], indent=2
+                            )  # pretty print the config in json
+                            print(msg)
```

Comment:
i thought of `stderr` too, it's helpful to be able to distinguish the streams. on the other hand it sounds a bit surprising.

if we were really wild we could issue ANSI codes to delete the preceding banner output, thus enabling `stdout` redirection. this seems like it might be a maintenance time-sink

maybe the most prudent solution is to wait for a use-case to emerge that prefers redirection? i don't have a wonderful solution to mind rn - rewriting configs in place is also hard to keep track of